File: Medium/0019-Remove Nth Node From End of List/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        dummy = ListNode(0)
        dummy.next = head
        node0 = head
        node1 = dummy

        for _ in range(n):
            node0 = node0.next

        while node0:
            node0 = node0.next
            node1 = node1.next

        node1.next = node1.next.next

        return dummy.next

File: Medium/0019-Remove Nth Node From End of List/Solution.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @param {Integer} n
# @return {ListNode}
def remove_nth_from_end(head, n)
    dummy = ListNode.new(0, head)
    node0 = head
    node1 = dummy

    for _ in 0...n
        node0 = node0.next
    end

    while node0
        node0 = node0.next
        node1 = node1.next
    end

    node1.next = node1.next.next

    return dummy.next
end
